component-catalog-connectors/example-components-connector/README.md

```diff
@@ -0,0 +1,52 @@
+## Elyra pipeline component examples catalog
+
+This catalog connector provides access to example pipeline components for Kubeflow Pipelines and Apache Airflow.
+
+### Install the component examples
+
+You can install the component examples from PyPI or source code. Note that a **rebuild of JupyterLab is not required**.
+
+**Prerequisites**
+
+- [Install Elyra](https://elyra.readthedocs.io/en/stable/getting_started/installation.html) (version 3.3 and above).
+
+**Install from PyPI**
+
+  ```
+  $ pip install elyra-example-components-catalog
+  ```
+
+**Install from source code**
+
+   ```
+   $ git clone https://github.com/elyra-ai/examples.git
+   $ cd examples/component-catalog-connectors/elyra-examples-connector/
+   $ make clean source-install
+   ```
+
+### Use the connector
+
+1. Launch JupyterLab.
+1. [Open the '`Manage Components`' panel](
+https://elyra.readthedocs.io/en/stable/user_guide/pipeline-components.html#managing-custom-components-using-the-jupyterlab-ui).
+1. Add a new component examples catalog ('`+`' > '`New Elyra component examples catalog`').
+1. Specify a catalog name, e.g. '`Elyra example components for Kubeflow`'.
```

Comment:
What is the behavior when a second instance of the same runtime is created?  Do users just see another set of operators?

Should there be a way to filter out components?  Could this be done by exposing the `file_filter` property in the schema?  (the problem there is that the default value would probably need to be `*` since it needs to span the runtime types, although it looks like kfp only has `yaml` and airflow only has `python`.)

Reply:
I think since the relative file paths would be the same for each instance, there won't be any duplication of components/nodes/operators because, if anything, the ones read/processed second will 'overwrite' (with the same information) the ones read/processed first

Reply:
Oh right.  I was thinking its the connector instance name that is used in the hash, but it's really the instance's underlying schema name.  (Is that correct?  Is the hashed op value <schema_name>:<hash of provided key values>?)

Regarding the filtering aspect, should users be able to filter on the component definition name?  This would allow them to distribute components across different categories.

Reply:
> (Is that correct? 

Yes!


> Regarding the filtering aspect

Several reasons why I'm leaning towards not supporting this (_for example connectors_ - but I am agreeing that this makes sense for generic packaged connectors):
 - We have a very small set of components to begin with.
 - Users won't know ahead of time what components are included and thus what tp filter by.

Also, we  really don't want to encourage users to use these examples for anything other than initial exploration. This was one concern I had with our initial approach: users must not assume that the same example components are available in _every_ Elyra release. For example, if we were to remove the spark operators in the future, all "production" pipelines that relied on those operators would fail. Users should therefore always only use components from catalogs that they have control over.